wordpress-mobile/WordPress-iOS
# Fix/site creation fixes

Fixes # N/A

This fixes misc Site Creation issues:
- The Site Creation views were rotating on iPhone after the controller refactor.
- The progress HUDs on Theme Selection and Domain Suggestions were not dismissed if 'Back' was pressed before the loading finished.
- The SiteType files needed to be renamed to fit the SiteCreation* naming convention.

To test:
- Verify the Site Creation views do not rotate on iPhone, and do rotate on iPad.
- On the Theme & Domain views, hit 'Back' before the loading is finished. Verify the HUDs are dismissed.
- If you see the first view, then the SiteType renaming worked. 😄
